packages/dappkit/src/index.ts

```diff
@@ -34,10 +45,52 @@ export function listenToAccount(callback: (account: string) => void) {
   })
 }
 
+export function waitForAccountAuth(requestId: string): Promise<DappKitResponse> {
+  return new Promise((resolve, reject) => {
+    const handler = ({ url }: { url: string }) => {
+      try {
+        const [dappKitResponse, returnedRequestId] = parseDappkitResponseDepplink(url)
+        if (
+          requestId === returnedRequestId &&
+          dappKitResponse.type === DappKitRequestTypes.ACCOUNT_ADDRESS &&
+          dappKitResponse.status === DappKitResponseStatus.SUCCESS
+        ) {
+          Linking.removeEventListener('url', handler)
```

Comment:
I don't think the event listener cleanup is super important, but it does seem that you will just have a hanging one if there is any sort of error or no response from the wallet

Reply:
Yeah I'm not a fan of this particular interface, ideally the original request resolves in a promise, but for now this works I'd say